src/xo-app/settings/remotes/index.js

```diff
@@ -80,25 +89,47 @@ class AbstractRemote extends Component {
       <td>{this._renderRemoteInfo(remote)}</td>
       <td>{this._renderAuthInfo(remote)}</td>
       <td>
-        {remote.enabled &&
-          <span>
-            <span className='text-success'>{this.accessible} <Icon icon='success' /></span>
-            {' '}
-            <Tooltip content={_('remoteTestTip')}><ActionRowButton btnStyle='primary' handler={this._test} icon='diagnosis' /></Tooltip>
-            {' '}
-            <Tooltip content={_('remoteDisconnectTip')}><ActionRowButton btnStyle='warning' handler={disableRemote} handlerParam={remote} icon='disconnect' /></Tooltip>
-          </span>
+        <StateButton
+          disabledLabel={_('remoteDisconnected')}
+          disabledHandler={enableRemote}
+          disabledTooltip={_('remoteConnectTip')}
+
+          enabledLabel={_('remoteConnected')}
+          enabledHandler={disableRemote}
+          enabledTooltip={_('remoteDisconnectTip')}
+
+          handlerParam={remote}
+          state={remote.enabled}
+          style={{
+            marginRight: '0.5em'
+          }}
+        />
+        {remote.error &&
+          <Tooltip content={_('remoteConnectionFailed')}>
+            <a
+              className='text-danger btn btn-link'
+              style={{ padding: '0px' }}
+              onClick={() => this._showError(remote.error)}
```

Comment:
Do not create a function here, access `this.props.remote` directly in the handler.